src/openforms/submissions/tests/test_submission_report.py

```diff
@@ -106,24 +269,73 @@ def test_report_is_generated_in_same_language_as_submission(self):
             submission__form__name_en="Translated form name",
             submission__form__formstep__form_definition__name_nl="Walsje",
             submission__form__formstep__form_definition__name_en="A Quickstep",
+            submission__form__formstep__form_definition__component_translations={
+                "en": translations
+            },
             submission__form__formstep__form_definition__configuration={
-                "components": [
-                    {
-                        "key": "foo",
-                        "type": "textfield",
-                        "label": "Foo",
-                        "showInEmail": True,
-                    }
-                ],
+                "components": components,
             },
         )
+        SubmissionStepFactory.create(
+            submission=report.submission,
+            form_step=report.submission.form.formstep_set.get(),
+            data=submission_data,
+        )
 
+        # create report
         html_report = report.generate_submission_report_pdf()
 
         self.assertIn("Translated form name", html_report)
         self.assertIn("A Quickstep", html_report)
         self.assertNotIn("Walsje", html_report)
 
+        # localized date, time and decimal point
+        self.assertIn("June 29, 1911", html_report)
+        self.assertIn("11:50 p.m.", html_report)
+        self.assertIn("1,100,000.00", html_report)
+
+        for component_type, user_input in fields:
+            with self.subTest(f"FormIO label for {component_type}"):
+                if component_type in ["date", "time", "currency"]:
+                    pass  # these are localized
+                else:
+                    self.assertIn(user_input, html_report)
+                self.assertNotIn(
+                    f"Untranslated {component_type.title()} label", html_report
+                )
+                self.assertIn(f"Translated {component_type.title()} label", html_report)
+
+        # assert structural labels
+        self.assertIn("Translated Field Set label", html_report)
+        self.assertNotIn(
+            "Translated Columns label", html_report
+        )  # 1451 -> never output a label
```

Comment:
I'd also add the assertion for the untranslated string since a bug in setting up the translations for this label would still pass this assertion.

Reply:
So did I :) It's a few lines down. I'll re-order the assertions so these two are adjacent.